typing/typemod.ml

```diff
@@ -1706,7 +1706,7 @@ and transl_signature ?(toplevel = false) env sg =
             typedtree, sg, final_env
         | Psig_attribute x ->
             Builtin_attributes.warning_attribute x;
-            if toplevel || not (Warnings.is_active (Misplaced_attribute ""))
+            if not (Warnings.is_active (Misplaced_attribute ""))
```

Comment:
Discarding attributes when they appear in a parsetree location where the warning `53` is disabled sounds useful in general. Maybe we could have a `filter` function  in `Buitin_attribute`? (and extend its use later on). Moreover, since the new code is not dependent on the toplevel argument, would it make sense to move it to the `warning_attribute` function ?

Reply:
Good idea bout merging this logic inside `warning_attribute`; done in https://github.com/ocaml/ocaml/pull/13170/commits/73c87206630c4d33d4f09c4a438da8f3a9ab9ff5. This fixes yet another small bug: in the following program,
```
class c =
  object
    [@@@warning "-53"]
    [@@@alert foo "foo"]
  end
```
the alert would trigger Warning 53 even though it the warning is supposed to be disabled. After moving the logic to `warning_attribute`, the warning is no longer triggered.

I added a few more tests to exhibit this behaviour.